pkg/providers/amifamily/bootstrap/bottlerocketsettings.go

```diff
@@ -115,5 +130,8 @@ func (c *BottlerocketConfig) MarshalTOML() ([]byte, error) {
 		c.SettingsRaw = map[string]interface{}{}
 	}
 	c.SettingsRaw["kubernetes"] = c.Settings.Kubernetes
+	if c.Settings.BootstrapCommands != nil {
+		c.SettingsRaw["bootstrap-commands"] = c.Settings.BootstrapCommands
```

Comment:
We'll always select bottlerocket images that support this, right? It looks like this was only added in the most recent OS tag.

Reply:
> What happens if an image references an older OS tag

This is a good callout. I checked with the BR folks and it seems like, at this point, there is definitely incompatibility with older versions of BR if we were to pass this into our userData script. There may be a way to check against the version of BR that we are launching or do something else fancier but I need to check with that team first.

Reply:
Moving this to draft for now since I think there's still some questions about how this works across BR versions.

Reply:
Ok, after more discussion, the maintainers agreed that it's reasonable to allow this configuration to get passed through on newer versions of Bottlerocket and newer versions of Karpenter. Specifying `instanceStorePolicy: RAID0` on older Bottlerocket images with older versions of Karpenter was invalid configuration.